hbase-server/src/main/java/org/apache/hadoop/hbase/io/hfile/HFileBlock.java

```diff
@@ -857,11 +859,17 @@ EncodingState getEncodingState() {
      * @param dataBlockEncoder data block encoding algorithm to use
      */
     public Writer(HFileDataBlockEncoder dataBlockEncoder, HFileContext fileContext) {
+      this(dataBlockEncoder, fileContext, ByteBuffAllocator.HEAP);
```

Comment:
Seems this constructor is only used for UT now ?  If so please mark it as VisiableForTesting and say only for testing purpose.. Or Just move it out of the Writer class (put it in UT class )?